channelserver/src/session.rs

```diff
@@ -48,13 +57,38 @@ impl Actor for WsChannelSession {
         // before processing any other events.
         // HttpContext::state() is instance of WsChatSessionState, state is shared
         // across all routes within application
-        let addr: Addr<Self> = ctx.address();
+
+        self.hb(ctx);
+
         self.meta = SenderData::from(ctx.request().clone());
+        // Check that the remote IP isn't abusive:
+        if let Some(remote) = self.meta.remote.clone() {
```

Comment:
I think "ref remote" kills this remote.clone() (and you might not need the & on line 70)